<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>937</width>
    <height>500</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>80</y>
     <width>111</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QCheckBox" name="saveValuesCheckBox">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>120</y>
     <width>131</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Save values?</string>
   </property>
  </widget>
  <widget class="QLabel" name="dateTimeLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>0</y>
     <width>141</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Date/time</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>30</y>
     <width>321</width>
     <height>181</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="LayoutRPM">
    <item>
     <widget class="QWidget" name="RPM" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QLineEdit" name="delay">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>170</y>
     <width>51</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="delay_label">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>150</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Time delay (ms)</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>230</y>
     <width>321</width>
     <height>181</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="LayoutPWM">
    <item>
     <widget class="QWidget" name="PWM" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QLineEdit" name="reference">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>170</y>
     <width>51</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="reference_label">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>150</y>
     <width>61</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Reference</string>
   </property>
  </widget>
  <widget class="QPushButton" name="StartStop">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>370</y>
     <width>111</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="manualinput">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>200</y>
     <width>121</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Manual input</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="automaticinput">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>340</y>
     <width>131</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Automatic input</string>
   </property>
  </widget>
  <widget class="QComboBox" name="tiposenal">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>260</y>
     <width>91</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QComboBox" name="modooperacion">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>40</y>
     <width>131</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Inhabilitado</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Identificaci贸n de Sistema</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Control de Velocidad</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Control de Posici贸n</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Modo Manual Velocidad</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Modo Manual Posici贸n</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="modooperacion_label">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>10</y>
     <width>141</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Modo de operaci贸n</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="tiemporeferencia">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>310</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="tiemporeferencia_label">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>290</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Reference time</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="amplitude">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>260</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="amplitude_label">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>240</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Amplitude</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="serial_in">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>430</y>
     <width>571</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="serial_out">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>470</y>
     <width>571</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="offset">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>310</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="offset_label">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>290</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Offset</string>
   </property>
  </widget>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>70</y>
     <width>381</width>
     <height>341</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <property name="iconSize">
    <size>
     <width>20</width>
     <height>12</height>
    </size>
   </property>
   <widget class="QWidget" name="tab">
    <attribute name="title">
     <string>Tab 1</string>
    </attribute>
    <widget class="QPushButton" name="identify">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>201</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Perform System Identification</string>
     </property>
    </widget>
    <widget class="QTextBrowser" name="identificationresult">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>201</width>
       <height>251</height>
      </rect>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>40</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_2">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>40</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="modooperacion_label_3">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>10</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>X lag            Y lag</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_3">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>100</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="modooperacion_label_4">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>70</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Degree</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_4">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>100</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="modooperacion_label_5">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>70</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Info values</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_3">
    <attribute name="title">
     <string>Page</string>
    </attribute>
    <widget class="QLineEdit" name="Kp">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>90</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Ki">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>120</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="Kd">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>150</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="Kp_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Kp</string>
     </property>
    </widget>
    <widget class="QLabel" name="Ki_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Ki</string>
     </property>
    </widget>
    <widget class="QLabel" name="Kd_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Kd</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="resettime">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>210</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lowpassfilter">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>240</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="resettime_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Tiempo de reset para evitar el windup integral&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>Tr</string>
     </property>
    </widget>
    <widget class="QLabel" name="lowpassfilter_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Constante de tiempo del filtro pasa baja para filtrado del ruido / patada derivativa&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>Tc</string>
     </property>
    </widget>
    <widget class="QLabel" name="modooperacion_label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>141</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;PID controller&lt;/p&gt;&lt;p&gt;(via bilinear transform)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="Digital">
    <attribute name="title">
     <string>Tab 2</string>
    </attribute>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Digital controller&lt;/p&gt;&lt;p align=&quot;center&quot;&gt;coefficients&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLabel" name="C_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>C</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="H">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>240</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="F_label">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>120</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>F</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="A">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="H_label">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>220</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>H</string>
     </property>
    </widget>
    <widget class="QLabel" name="B_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>B</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="G">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>190</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="C">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>190</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="F">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>140</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="D_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>220</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>D</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="D">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>240</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="G_label">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>170</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>G</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="E">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>90</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="B">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="E_label">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>70</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>E</string>
     </property>
    </widget>
    <widget class="QLabel" name="A_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>A</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QSlider" name="slider">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>50</y>
     <width>21</width>
     <height>141</height>
    </rect>
   </property>
   <property name="maximum">
    <number>255</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QLabel" name="tiposenal_label">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>240</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Signal type</string>
   </property>
  </widget>
  <widget class="QLabel" name="reference_label_2">
   <property name="geometry">
    <rect>
     <x>140</x>
     <y>200</y>
     <width>61</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Manual ref</string>
   </property>
  </widget>
  <widget class="QPushButton" name="update_parameters">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>430</y>
     <width>131</width>
     <height>61</height>
    </rect>
   </property>
   <property name="text">
    <string>Update parameters</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
